### PR TITLE
[Enhancement] Change Xmx in start_fe to 8G

### PR DESCRIPTION
**Pull request by gengjun-git**
## Problem Summary:
Fixes # (issue)
If there are some Illegal characters in the fe.conf, the JAVA_OPTS will be read in the start_fe.sh, but the Xmx in start_fe.sh is 1G, which will cause OOM.
Set Xmx in start_fe.sh to 8G.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: bin/start_fe.sh [hunk 1]
```diff
@@ -60,7 +60,7 @@ source $STARROCKS_HOME/bin/common.sh
 # JAVA_OPTS
 # LOG_DIR
 # PID_DIR
-export JAVA_OPTS="-Xmx1024m"
+export JAVA_OPTS="-Xmx8G"
 export LOG_DIR="$STARROCKS_HOME/log"
```
Comment (kevincai) [inline bin/start_fe.sh:64]: -Xmx8g?

wondering if we can do more in common.sh, e.g. check host mem, allocate half of total mem up to 8g at most by default.

File: bin/start_fe.sh [hunk 2]
```diff
@@ -60,7 +60,7 @@ source $STARROCKS_HOME/bin/common.sh
 # JAVA_OPTS
 # LOG_DIR
 # PID_DIR
-export JAVA_OPTS="-Xmx1024m"
+export JAVA_OPTS="-Xmx8G"
```
Comment (kevincai) [inline bin/start_fe.sh:63]: -Xmx8g?

wondering if we can do more in common.sh, e.g. check host mem, allocate half of total mem up to 8g at most by default.

Comment (gengjun-git) [reply bin/start_fe.sh]: Good advice. Maybe we achieve it in the following PR.

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**